File: CS109/Lab1/rpnc.py
#
# Python3 Reverse Polish Notation (RPN) Calculator
#
# Accepts an single argument in RPN, evaluates it and prints the answer to stdout.
# 
if __name__ == '__main__':
	stack = []
	string = str(input())
	#http://www.pythonforbeginners.com/dictionary/python-split
	array = string.split(":")
	print(array[0])
	array = array[0].split(" ")
	for word in array:
		
		#https://stackoverflow.com/questions/5424716/how-to-check-if-string-input-is-a-number
		if(word.isdigit()):
			stack.append(int(word))
		
		elif(word != " " and word != ":"):
			b = stack.pop()
			a = stack.pop()
			result = 0

			if(word == '+'):
				result = a + b
			elif(word == '-'):
				result = a - b
			elif(word == '/'):
				result = a / b
			elif(word == '*'):
				result = a * b
			elif(word == '^'):
				result = a ** b
			stack.append(result)

	print(int(stack.pop()))

File: CS109/Lab10/rpnc.js
/*
 * Node.js / JavaScript Reverse Polish Notation (RPN) Calculator
 *
 * Accepts an single argument in RPN, evaluates it and prints the answer to stdout.
 * 
 */

//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions
var CalcRPNC = function(needsRPNC)
{
	var stack = [];
	for (var i = 0; i < needsRPNC.length; i++)
	{
		var char = needsRPNC[i];
		//console.log(char);
		if(char == '*')
		{
			var intOne = stack.pop();
			var intTwo = stack.pop();
			var result = intTwo*intOne;
			//console.log(intTwo + " * " + intOne + " = " + result);
			stack.push(result);
		}
		else if(char == "-")
		{
			var intOne = stack.pop();
			var intTwo = stack.pop();
			var result = intTwo-intOne;
			//console.log(intTwo + " - " + intOne + " = " + result);
			stack.push(result);
		}
		else if(char == "+")
		{
			var intOne = stack.pop();
			var intTwo = stack.pop();
			var result = intTwo+intOne;
			//console.log(intTwo + " + " + intOne + " = " + result);
			stack.push(result);
		}
		else if(char == "/")
		{
			var intOne = stack.pop();
			var intTwo = stack.pop();
			var result = intTwo/intOne;
			//console.log(intTwo + " / " + intOne + " = " + result);
			stack.push(result);
		}
		else if(char == "^")
		{
			var intOne = stack.pop();
			var intTwo = stack.pop();
			var result = Math.pow(intTwo, intOne);
			//console.log(intTwo + " ^ " + intOne + " = " + result);
			stack.push(result);
		}
		else
		{
			//console.log(char);
			var num = parseInt(char); 
			stack.push(num);
		}
	}
	return stack[0];
}

//https://stackoverflow.com/questions/4351521/how-do-i-pass-command-line-arguments-to-a-nodejs-program
var needsRPNC = process.argv[2].split(" ");
var done = CalcRPNC(needsRPNC);
console.log(done);
